apps/wallet-mobile/src/features/SetupWallet/useCases/RestoreHwWallet/ConnectNanoXScreen.tsx

```diff
@@ -30,16 +35,45 @@ type Props = {
 export const ConnectNanoXScreen = ({defaultDevices}: Props) => {
   const intl = useIntl()
   const strings = useStrings()
-  const styles = useStyles()
+  const {styles} = useStyles()
+  const {walletManager} = useWalletManager()
+  const {openModal} = useModal()
+  const {resetToTxHistory} = useWalletNavigation()
   const navigation = useNavigation<SetupWalletRouteNavigation>()
 
   const {hwDeviceInfoChanged, walletImplementation, useUSB} = useSetupWallet()
 
   const onSuccess = (hwDeviceInfo: HW.DeviceInfo) => {
     hwDeviceInfoChanged(hwDeviceInfo)
+    const plate = walletChecksum(hwDeviceInfo.bip44AccountPublic)
+
+    const duplicatedWalletMeta = Array.from(walletManager.walletMetas.values()).find(
```

Comment:
@banklesss on top of my head, it looks like this should be part of WalletManager API, something like `isWalletNameValid`, and then it checks if the constrains e.g size, available characters and also if the name already exist or not, because when calling `renameWallet` all these checks should happen again, so it should throw `DuplicatedName` `InvalidName`.

Reply:
Thanks for the suggestion! I have updated the wallet manager with the new functions which are more atomic because the different checks are done on different screens of the flow

Please tell me if you think it could be improved!